### PR TITLE
Add inline vars

### PR DESCRIPTION
**Pull request by frogonwheels**
Add support for new delphi inline variables.
 (also updates to latest binding interface)

### CODE REVIEW

**Comment by frogonwheels** [comment]
Just realised I missed out 
`for var i := 0 to 10 do`
so wait for that.

**Comment by frogonwheels** [comment]
Added a test for inline variable in for loop.  Was working fine.

File: grammar.js [hunk 1]
```diff
@@ -119,6 +119,15 @@ function statements(trailing) {
...
 	return Object.fromEntries([
+		[rn('var'),         $ => seq(
+			$.kVar,
+			$.identifier,
+			optional(seq(
+				':',
+				field('type', $.typeref)
+			)),
+			optional(seq($.kAssign, $._expr)
+		))],
```
Comment (Isopod) [inline grammar.js:130]: Is this part really needed? The node doesn't seem to be used anywhere in the tests.

**Review by Isopod — COMMENTED** [review]
Thanks for the contribution! I tested it and it seems to work fine, however, I think there is a redundant section. Also, I apologize for the test failure, didn't notice it until now, I hope it didn't cause any confusion. I'm gonna fix that later.

**Pull request merged**